### ISSUE TITLE
Use Rye for package dependencies

### ISSUE THREAD

**Issue by Nitnelav** [Universite-Gustave-Eiffel/acoustic-toolbox#24]
Looks good i'll switch to that !

From what I've red we need to remove requirements.txt and rely on pyproject.toml for dependencies.
I believe `requirements(-dev).lock` files should be committed though

**Comment by MitchellAcoustics** [comment]
@Nitnelav I know we just changed to Rye on my recommendation, but recent updates to [uv](https://docs.astral.sh/uv/) have integrated the most important features. Rye's documentation notes: 

> If you're getting started with Rye, consider [uv](https://github.com/astral-sh/uv), the [successor project](https://lucumr.pocoo.org/2024/2/15/rye-grows-with-uv/) from the same maintainers.

> While Rye is actively maintained, uv offers a more stable and feature-complete experience, and is the recommended choice for new projects.

So for stability, more documentation, and support, we should probably migrate now while it's still early in our project.

Switching should be very easy, since Rye was using uv behind the scenes anyway and should already be installed for anyone using Rye. I'll get started on it now. 

Sorry for being just a little bit early! They released the version of uv with Rye features integrated like a month after I suggested it.
